ripley/lib/canvas_site_utils.py

```diff
@@ -215,30 +210,32 @@ def get_teaching_terms(current_user=None, section_ids=None, sections=None, term_
     elif current_user and (current_user.is_teaching or current_user.canvas_masquerading_user_id):
         instructor_uid = current_user.uid
 
-    teaching_sections = []
-    if instructor_uid:
-        teaching_sections = data_loch.get_instructing_sections(instructor_uid, [t.to_sis_term_id() for t in terms]) or []
-        teaching_sections = sort_course_sections(teaching_sections)
-
-    if not len(teaching_sections) and sections:
-        teaching_sections = sections
+    term_ids = [t.to_sis_term_id() for t in terms]
+    teaching_sections = data_loch.get_instructing_sections(instructor_uid, term_ids) if instructor_uid else []
+    if sections:
+        for section in sections:
+            section_id = section['section_id']
+            term_id = section['term_id']
+            match = next((s for s in teaching_sections if s['term_id'] == term_id and s['section_id'] == section_id), None)
+            if not match:
+                teaching_sections.append(section)
```

Comment:
The fix is above. We include `sections` (see incoming param) not represented in `teaching_sections`.